### PR TITLE
abscomponent: added repr_alis()

### PR DESCRIPTION
**Pull request by ntejos**
abscomponent: added repr_alis()

### CODE REVIEW

File: linetools/isgm/abscomponent.py [hunk 1]
```diff
@@ -430,11 +430,15 @@ def repr_vpfit(self, b=10.):
         Parameters
         ----------
         b : float, optional
-            Doppler parameter of the component
+            Doppler parameter of the component (km/s)
```
Comment (profxj) [inline linetools/isgm/abscomponent.py]: Is b a float or a Quantity?
Seems like it should be the latter.

File: linetools/isgm/abscomponent.py [hunk 2]
```diff
@@ -430,11 +430,15 @@ def repr_vpfit(self, b=10.):
...
+            Default is 10.
 
         Returns
         -------
         repr_vpfit : str
+
+
+        [todo: consider add optional `tie` and `fix` keys]
```
Comment (profxj) [inline linetools/isgm/abscomponent.py]: yes, tie and fix will be very useful.

File: linetools/isgm/abscomponent.py [hunk 3]
```diff
@@ -445,8 +449,47 @@ def repr_vpfit(self, b=10.):
...
+        s += '\n'
         return s
 
+    def repr_alis(self, T_kin=1e4,bturb=0.):
+        """
+        String representation for ALIS
+
+        Parameters
+        ----------
+        T_kin : float, optional
```
Comment (profxj) [inline linetools/isgm/abscomponent.py]: Again, a Quantity makes more sense here.

I like using them as often as I can.  Can be
cumbersome but is likely to save you in the long run.

File: linetools/isgm/abscomponent.py [hunk 4]
```diff
@@ -445,8 +449,47 @@ def repr_vpfit(self, b=10.):
...
+            Kinetic temperature (K)
+            Default is 1e4
+        bturb : float, optional
```
Comment (profxj) [inline linetools/isgm/abscomponent.py]: Same.

File: linetools/isgm/abscomponent.py [hunk 5]
```diff
@@ -445,8 +449,47 @@ def repr_vpfit(self, b=10.):
...
+        Returns
+        -------
+        repr_alis : str
+
+
+        [todo: consider add optional `tie` and `fix` keys]
+        """
+
+        # A patch for nucleons; todo: come up with a better way to do this using ELEMENTS?
+        if self.Zion[0]==1:
```
Comment (profxj) [inline linetools/isgm/abscomponent.py]: spaces between operators and other things (PEP)

File: linetools/isgm/abscomponent.py [hunk 6]
```diff
@@ -445,8 +449,47 @@ def repr_vpfit(self, b=10.):
...
+            nucleons = 1
+        elif self.Zion[0]>1:
+            nucleons = 2*self.Zion[0]
+
+        name = '{}'.format(nucleons)+self.name.replace(' ','_')
```
Comment (profxj) [inline linetools/isgm/abscomponent.py]: Is there a reason not to specify the format type, i.e. {:d}?

File: linetools/isgm/abscomponent.py [hunk 7]
```diff
@@ -445,8 +449,47 @@ def repr_vpfit(self, b=10.):
...
+
+        try:
```
Comment (profxj) [inline linetools/isgm/abscomponent.py]: self.logN is always set now.
Remove.

File: linetools/isgm/abscomponent.py [hunk 8]
```diff
@@ -445,8 +449,47 @@ def repr_vpfit(self, b=10.):
...
+            logN = self.logN
+        except:
+            logN = 0.
+        s = 'voigt   ion={} {:.2f} redshift={:.5f} {:.2f} {:.2f}'.format(name,logN,self.zcomp,bturb,T_kin)
+        if len(self.comment)>0:
+            s += '# {}'.format(self.comment)
```
Comment (profxj) [inline linetools/isgm/abscomponent.py]: Same thing on formatting.  Use  {:s}

**Comment by profxj** [comment]
Address those and merge.

And I was hoping to grab PR#100! :)

File: linetools/isgm/abscomponent.py (hunk 1 above)
Comment (ntejos) [inline linetools/isgm/abscomponent.py]: Will do.

File: linetools/isgm/abscomponent.py (hunk 2 above)
Comment (ntejos) [inline linetools/isgm/abscomponent.py]: Done in a way I found reasonable.

File: linetools/isgm/abscomponent.py (hunk 3 above)
Comment (ntejos) [inline linetools/isgm/abscomponent.py]: Done.

File: linetools/isgm/abscomponent.py (hunk 6 above)
Comment (ntejos) [inline linetools/isgm/abscomponent.py]: {:d} is the default, but will specify it explicitly now.

File: linetools/isgm/abscomponent.py (hunk 7 above)
Comment (ntejos) [inline linetools/isgm/abscomponent.py]: Will need to add it to the main class to pass the current tests.

**Comment by profxj** [comment]
merging.

**Comment by profxj** [comment]
p.s.  Can you update the docs too?

**Pull request merged**